internals/plan/plan.go

```diff
@@ -32,6 +32,31 @@ import (
 	"github.com/canonical/pebble/internals/osutil"
 )
 
+// LayerSectionExtension allows the plan layer schema to be extended without
+// adding centralised schema knowledge to the plan library.
+type LayerSectionExtension interface {
+	// ParseSection returns a newly allocated concrete type containing the
+	// unmarshalled section content.
+	ParseSection(data yaml.Node) (LayerSection, error)
+
+	// CombineSections returns a newly allocated concrete type containing the
+	// result of combining the supplied sections in order.
+	CombineSections(sections ...LayerSection) (LayerSection, error)
+
+	// ValidatePlan takes the complete plan as input, and allows the
+	// extension to validate the plan. This can be used for cross section
+	// dependency validation.
+	ValidatePlan(plan *Plan) error
+}
+
+type LayerSection interface {
```

Comment:
Question from Ben: Could this simply become ```plan.Section``` ?

Reply:
Applied change to ```plan.Section```. It also feels a bit more accurate as sections to appear directly in the Plan, not only a layer.

Reply:
Resolving.